### PR TITLE
[DataProtectorSharing SC] Integrate iExec account

### PR DESCRIPTION
**Pull request by Le-Caignec**

### CODE REVIEW

File: packages/sharing-smart-contract/contracts/registry/AppWhitelist.sol
Comment (PierreJeanjacquot) [inline packages/sharing-smart-contract/contracts/registry/AppWhitelist.sol:1]: I suggest renaming the contract to `AddOnlyAppWhitelist` to explicit that it is not a standard whitelist

File: packages/sharing-smart-contract/.prettierrc [hunk 1]
```diff
@@ -1,9 +1,9 @@
 {
-  "printWidth": 100,
+  "printWidth": 120,
```
Comment (PierreJeanjacquot) [inline packages/sharing-smart-contract/.prettierrc:2]: changing the print width made the PR much more difficult to read :grimacing: 
next time avoid doing this in a PR not dedicated to code formatting

**Review by PierreJeanjacquot — APPROVED** [review]
The PR has a lot of noise due to the change of the formatting config, nevertheless it looks good, I hope I didn't miss anything critical.
I left a few comments that might be good to address.

File: packages/sharing-smart-contract/contracts/DataProtectorSharing.sol [hunk 2]
```diff
@@ -115,62 +111,50 @@ contract DataProtectorSharing is
     function _checkConsumeProtectedData(
```
[the remark this answers is not in the document]
Comment (Le-Caignec) [reply packages/sharing-smart-contract/contracts/DataProtectorSharing.sol]: Commit: e143d03b7560f24351e0383302fc81bca9184206

File: packages/sharing-smart-contract/contracts/registry/AppWhitelist.sol
Comment (Le-Caignec) [reply packages/sharing-smart-contract/contracts/registry/AppWhitelist.sol:1]: Commit: 7321f78422f334f233771bb21c743cd7b5425bfd

**Pull request merged**